specs/commandline/blob.md

```diff
@@ -0,0 +1,501 @@
+{
+    "version": "1.0",
+    "trustPolicies": [
+        {
+            "name": "wabbit-networks-policy",
+            "signatureVerification": {
+                "level": "strict"
+            },
+            "trustStores": [ 
+              "ca:wabbit-networks",
+            ],
+            "trustedIdentities": [
+                "x509.subject: C=US, ST=WA, L=Seattle, O=wabbit-networks.io, OU=Security Tools"
+            ]
+        },
+        {
+            "name": "skip-verification-policy",
+            "signatureVerification": {
+              "level" : "skip" 
+            }
+        },
+        {
+            "name": "global-verification-policy",
+            "globalPolicy": true,
+            "signatureVerification": {
+              "level" : "audit"
+            },
+            "trustStores": ["ca:acme-rockets"],
+            "trustedIdentities": ["*"]
+        }
+    ]
+}
+```
+
+## Outline
+
+### notation blob command
+
+```text
+Sign, inspect, and verify signatures and configure trust policies.
+
+Usage:
+  notation blob [command]
+
+Available Commands:
+  inspect   inspect a signature associated with a blob
+  policy    manage trust policy configuration for signed blobs
+  sign      produce a detached signature for a given blob
+  verify    verify a signature associated with a blob
+
+Flags:
+  -h, --help   help for blob
+```
+
+### notation blob sign
+
+```text
+Produce a signature for a given blob. A detached signature file will be written to the currently working directory with blob file name + ".sig" + signature format as the file extension. For example, signature file name for "myBlob.bin" will be "myBlob.bin.sig.jws" for JWS signature format or "myBlob.bin.sig.cose" for COSE signature format.
+
+Usage:
+  notation blob sign [flags] <blob_path>
+
+Flags:
+       --signature-directory string optional path where the blob signature needs to be placed (default: currently working directory) 
+       --media-type string          optional media type of the blob (default: "application/octet-stream")
+  -e,  --expiry duration            optional expiry that provides a "best by use" time for the blob. The duration is specified in minutes(m) and/or hours(h). For example: 12h, 30m, 3h20m
+       --id string                  key id (required if --plugin is set). This is mutually exclusive with the --key flag
+  -k,  --key string                 signing key name, for a key previously added to notation's key list. This is mutually exclusive with the --id and --plugin flags
+       --plugin string              signing plugin name. This is mutually exclusive with the --key flag
+       --plugin-config stringArray  {key}={value} pairs that are passed as it is to a plugin, refer plugin's documentation to set appropriate values.
+       --signature-format string    signature envelope format, options: "jws", "cose" (default "jws")
+       --force                      skip user confirmation and force overwrite the existing blob signature file 
```

Comment:
What's the real-world scenario of overwriting an existing blob signature file?

Reply:
An example : User provided wrong media-type or plugin first time and they want to correct it by re-running the `sign` command.

Reply:
@shizhMSFT Do you have any follow up questions on this? If not, can you please approve the PR?

Reply:
> User provided wrong media-type or plugin first time and they want to correct it by re-running the sign command.

In that scenario, customers should delete the previous signature securely. Instead of overwriting, customers should use `shred` command to securely delete it (so that it cannot be recovered from file system) or even revoke the signature if it is already spread.

Reply:
What is your recommendation here?

Reply:
@shizhMSFT ^

Reply:
I'd recommend treating "overwriting the existing blob signature file" as an error.

@priteshbandi @yizha1 @FeynmanZhou We also need a doc for guidance in case the customer mis-signed an artifact or a blob.

Reply:
I think this requires more conversation. I removed the "overwriting" feature for now. We can add it back if needed in the future.